### PR TITLE
octo-logger-cpp/v1.2.0 - AWS support

### PR DESCRIPTION
**Pull request by ofiriluz**
Specify library name and version:  **octo-logger-cpp/1.2.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

octo-logger-cpp/v1.2.0 - AWS support

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/octo-logger-cpp/all/conanfile.py [hunk 1]
```diff
@@ -28,6 +34,14 @@ def _compilers_minimum_version(self):
             "msvc": "1923",
         }
 
+    @property
+    def _aws_supported(self):
+        return Version(self.version) >= "1.2.0"
+
+    def configure(self):
+        if self.options.with_aws and self._aws_supported:
+            self.options["aws-sdk-cpp"].logs = True
```
Comment (uilianries) [inline recipes/octo-logger-cpp/all/conanfile.py:43]:
```suggestion
            self.options["aws-sdk-cpp"].logs = True
        if not self._aws_supported:
            del self.options.with_aws
```
Older version should not support it.

Comment (uilianries) [inline recipes/octo-logger-cpp/all/conanfile.py:43]:
```suggestion
        if not self._aws_supported:
            del self.options.with_aws
```

Older versions should not have that option.
Move the mandatory dependency option to validate() method:


```
def validate(self):
    if self.info.options.get_safe("with_aws"):
        if not self.dependencies["aws-sdk-cpp"].options.logs:
            raise ConanInvalidConfiguration(f"{self.name} requires the option aws-sdk-cpp:logs=True")
```

File: recipes/octo-logger-cpp/all/conanfile.py [hunk 2]
```diff
@@ -61,13 +75,21 @@ def source(self):
 
     def requirements(self):
         self.requires("fmt/9.0.0")
+        if self.options.with_aws and self._aws_supported:
```
Comment (uilianries) [inline recipes/octo-logger-cpp/all/conanfile.py:78]:
```suggestion
        if self.options.get_safe("with_aws"):
```

File: recipes/octo-logger-cpp/all/conanfile.py [hunk 3]
```diff
@@ -61,13 +75,21 @@ def source(self):
...
 
     def build(self):
         cmake = CMake(self)
-        cmake.configure()
+        variables = {}
+        if self._aws_supported:
+            variables.update({
+                "WITH_AWS": self.options.with_aws
+            })
+        cmake.configure(variables=variables)
```
Comment (uilianries) [inline recipes/octo-logger-cpp/all/conanfile.py:92]:
```suggestion
        cmake.configure()
```
It should be done in CMakeToolchain

File: recipes/octo-logger-cpp/all/conanfile.py [hunk 4]
```diff
@@ -81,6 +103,11 @@ def package_info(self):
         self.cpp_info.set_property("pkg_config_name", "octo-logger-cpp")
         self.cpp_info.components["libocto-logger-cpp"].libs = ["octo-logger-cpp"]
         self.cpp_info.components["libocto-logger-cpp"].requires = ["fmt::fmt"]
+        if self.options.with_aws and self._aws_supported:
```
Comment (uilianries) [inline recipes/octo-logger-cpp/all/conanfile.py:106]:
```suggestion
        if self.options.get_safe("with_aws"):
```

Comment (ofiriluz) [reply recipes/octo-logger-cpp/all/conanfile.py]: fixed

File: recipes/octo-logger-cpp/all/conanfile.py (hunk 3 above)
Comment (ofiriluz) [reply recipes/octo-logger-cpp/all/conanfile.py]: gotcha, fixed

File: recipes/octo-logger-cpp/all/conanfile.py (hunk 1 above)
Comment (ofiriluz) [reply recipes/octo-logger-cpp/all/conanfile.py]: gotcha, fixed both

**Comment by uilianries** [comment]
@ofiriluz `self.info.options.get_safe` is available on Conan 1.51, but we still are running 1.50 in CCI. Let's keep `self.options.get_safe`, we can improve it in the future.

**Review by uilianries — APPROVED** [review]

**Review by ericLemanissier — APPROVED** [review]

**Pull request merged**